common/src/main/java/io/pravega/common/util/ByteArraySegment.java

```diff
@@ -169,6 +171,45 @@ public void copyTo(OutputStream stream) throws IOException {
         stream.write(this.array, this.startOffset, this.length);
     }
 
+    @Override
+    @SneakyThrows(IOException.class)
+    public boolean equals(BufferView other) {
```

Comment:
See my comment on AbstractBufferView, (I think we can eliminate the need for this)